src/Raven.Server/Documents/ETL/Providers/Queue/AwsSqs/AwsSqsEtl.cs

```diff
@@ -0,0 +1,270 @@
+                    {
+                        QueueUrl = _alreadyCreatedQueues.GetValueOrDefault(queueName),
+                        MessageBody = message.MessageBody
+                    };
+
+                    AsyncHelpers.RunSync(() => _queueClient.SendMessageAsync(sendMessageRequest));
+
+                    if (queue.DeleteProcessedDocuments)
+                    {
+                        idsToDelete.Add(message.Id);
+                    }
+                }
+                catch (AmazonSQSException sqsEx)
+                {
+                    if (sqsEx.ErrorCode == "InvalidAttributeValue")
+                    {
+                        tooLargeDocsErrors.Enqueue(new EtlErrorInfo()
+                        {
+                            Date = DateTime.UtcNow, DocumentId = message.Id, Error = sqsEx.Message
+                        });
+                    }
+                    else
+                    {
+                        throw new QueueLoadException(
+                            $"Failed to deliver message, Aws error code: '{sqsEx.ErrorCode}', error reason: '{sqsEx.Message}' for document with id: '{message.Id}'",
+                            sqsEx);
+                    }
+                }
+                catch (Exception innerEx)
+                {
+                    throw new QueueLoadException($"Failed to deliver message, error reason: '{innerEx.Message}'",
+                        innerEx);
+                }
+            }
+        }
+    }
+
+
+    private string SerializeCloudEvent(AwsSqsItem queueItem)
+    {
+        var cloudEvent = CreateCloudEvent(queueItem);
+        return JsonSerializer.Serialize(cloudEvent, JsonSerializerOptions);
+    }
+
+    protected override void OnProcessStopped()
+    {
+        _queueClient?.Dispose();
+        _queueClient = null;
+        _alreadyCreatedQueues.Clear();
+    }
+
+    private async Task CreateQueue(IAmazonSQS queueClient, string queueName)
+    {
+        try
+        {
+            CreateQueueResponse createQueueResponse = await queueClient.CreateQueueAsync(queueName);
+            _alreadyCreatedQueues.Add(queueName, createQueueResponse.QueueUrl);
+
+            try
+            {
+                await queueClient.GetQueueUrlAsync("connection-test");
+            }
+            catch (AmazonSQSException e)
+            {
+                Console.WriteLine(e);
+                throw;
+            }
+            
+            
+
+            // we must wait at least one second after the queue is created to be able to use the queue
+            await Task.Delay(TimeSpan.FromMilliseconds(1000));
+        }
+        catch (AmazonSQSException ex)
+        {
+            throw new QueueLoadException(
+                $"Failed to create queue, Aws error code: '{ex.ErrorCode}', error reason: '{ex.Message}'", ex);
+        }
+    }
+
+    private sealed class CloudEventConverter : JsonConverter<CloudEvent>
```

Comment:
Don't we have already that class defined somewhere? Can we avoid code duplication?

Reply:
fixed